conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -90,22 +93,24 @@ def content(self):
 
             if dep.new_cpp_info.has_components:
                 components = self._get_components(dep)
+                # Adding one *.pc file per component, e.g., pkg-comp1.pc
                 for comp_genname, comp_cpp_info, comp_requires_gennames in components:
-                    pkg_comp_genname = "%s-%s" % (pkg_genname, comp_genname)
+                    pkg_comp_genname = self._get_composed_require_name(pkg_genname, comp_genname)
                     ret["%s.pc" % pkg_comp_genname] = self._pc_file_content(
                         pkg_comp_genname, comp_cpp_info,
-                        comp_requires_gennames, dep.package_folder,
-                        dep.ref.version)
+                        comp_requires_gennames,
+                        dep.package_folder, dep.ref.version)
+                # Adding the pkg *.pc file (including its components as requires if any)
                 comp_gennames = [comp_genname for comp_genname, _, _ in components]
                 if pkg_genname not in comp_gennames:
+                    pkg_requires = (self._get_composed_require_name(pkg_genname, i)
+                                    for i in comp_gennames)
                     ret["%s.pc" % pkg_genname] = self._global_pc_file_contents(pkg_genname,
                                                                                dep,
-                                                                               comp_gennames)
+                                                                               pkg_requires)
             else:
-                require_public_deps = [_d for _, _d in
-                                       self._get_public_require_deps(dep)]
                 ret["%s.pc" % pkg_genname] = self._pc_file_content(pkg_genname, dep.new_cpp_info,
-                                                                   require_public_deps,
+                                                                   self._get_public_require_deps(dep),
```

Comment:
This looks like was getting a list before, and now is going to receive tuples. Or at least the change is not evident.

Reply:
We were getting a list of tuples before, and now a list of strings. I think it's clearer now because we were not using all the values returned.